app/models/event_schedule.rb

```diff
@@ -31,6 +31,10 @@ def intersecting_events
     room.event_schedules.where(start_time: start_time, schedule: schedule).where.not(id: id)
   end
 
+  def replacement?
+    event.state == 'confirmed' && (!intersecting_events.canceled.empty? || !intersecting_events.withdrawn.empty?)
```

Comment:
This should be the same as:
`event.state == 'confirmed' && intersecting_events.present?`

right? because we shouldn't have several confirmed events scheduled at the same time.